pyiri90/runiri90.py

```diff
@@ -18,12 +18,18 @@ def runiri(dt,z,glat,glon,f107,f107a,ap,mass=48):
     JF = array((1,1,1,1,0,1,1,1,1,1,1,0),bool) #Solomon 1993 version of IRI
     #JF = (1,1,1) + (0,0,0) +(1,)*14 + (0,1,0,1,1,1,1,0,0,0,1,1,0,1,0,1,1,1) #for 2013 version of IRI
 #%% call IRI
-    chdir(pyiri90.__path__[0])
-    logging.debug(getcwd())
-    outf,oarr = iri90.iri90(JF,jmag,glat,glon % 360., -f107,
-                       dt.strftime('%m%d'),
-                       dt.hour+dt.minute//60+dt.second//3600,
-                       z,'data/')
+    cwd = getcwd()
+    try:
+        chdir(pyiri90.__path__[0])
+        logging.debug(getcwd())
+        outf,oarr = iri90.iri90(JF,jmag,glat,glon % 360., -f107,
+                        dt.strftime('%m%d'),
+                        dt.hour+dt.minute//60+dt.second//3600,
+                        z,'data/')
+    except Exception as e:
+        chdir(cwd)
+        raise e
+    chdir(cwd)
```

Comment:
This can be simplified to:
```python
    finally:
        chdir(cwd)
```

Reply:
Right, `finally` is so much easier. I use Python exceptions coming from a very C++ paradigm lol

I believe upstream iri90 has been using executable binaries lately and using subprocess + stdout capture to do things. Not sure I like that approach. I will make those changes. Thanks man.

Reply:
and I just applied the fixes.